### PR TITLE
Reworks VaRA documentation

### PR DESCRIPTION
**Pull request by vulder**
Migrates old docs from VaRA readme to docs and updates the information.
Adds an overview of the tool suite.
Adds VaRA logo to VaRA docs page.

resolves se-passau/VaRA#593
resolves se-passau/VaRA#570
resolves se-passau/VaRA#567

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -8,20 +8,24 @@ Welcome to VaRA's documentation!
...
 
 .. toctree::
    :maxdepth: 1
    :caption: List of provided research tools:
 
-   vara-ts-api/tools/research_tools/vara
+   research_tool_docs/vara/vara
+
+
+Tool-suite overview
```
Comment (cormensratio) [inline docs/source/index.rst]:
```suggestion
Tool-Suite overview
```

File: docs/source/research_tool_docs/vara/setup.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  git submodule init && git submodule update --recursive
+
+Second, checkout the VaRA repository as ``vara`` into ``vara-llvm-project``.
+
+.. code-block:: bash
+
+  git clone https://github.com/se-passau/VaRA.git vara
+  git submodule init && git submodule update --recursive
+
+Third, to complet the setup link the prepared VaRA build scripts into a build folder.
```
Comment (cormensratio) [inline docs/source/research_tool_docs/vara/setup.rst]:
```suggestion
Third, to complete the setup link the prepared VaRA build scripts into a build folder.
```

File: docs/source/vara-ts/benchbuild.rst [hunk 3]
```diff
@@ -22,3 +22,57 @@ Second, we change into the benchbuild folder and run an experiment that generate
   benchbuild -vv run -E GenerateBlameReport gzip
 
 The generated result files are place in the ``vara/results/$PROJECT_NAME`` folder and can be further visualized with VaRA-TS graph generators.
+
+How-to configure BenchBuild yourself
+------------------------------------
+BenchBuilds configuration file ``.benchbuild.yml`` normaly is placed inside the `benchbuild` folder, which is located in the vara root folder.
```
Comment (cormensratio) [inline docs/source/vara-ts/benchbuild.rst]:
```suggestion
BenchBuild's configuration file ``.benchbuild.yml`` normally is placed inside the `benchbuild` folder, which is located in the vara root folder.
```

File: docs/source/vara-ts/benchbuild.rst [hunk 4]
```diff
@@ -22,3 +22,57 @@ Second, we change into the benchbuild folder and run an experiment that generate
...
+A default version of this file can be automatically generated with our tool :ref:`vara-gen-bbconfig`.
+To adapt and tune BenchBuild further a lot of different configuration flags can be adapted, the following list shows the most important ones:
```
Comment (cormensratio) [inline docs/source/vara-ts/benchbuild.rst]:
```suggestion
To adapt and tune BenchBuild further a lot of different configuration flags can be adapted. The following list shows the most important ones:
```

File: docs/source/vara-ts/benchbuild.rst [hunk 5]
```diff
@@ -22,3 +22,57 @@ Second, we change into the benchbuild folder and run an experiment that generate
...
+
+* Adding extra paths to the environment
+
+.. code-block:: yaml
+
+  env:
+    path:
+      value: ["paths from your system that should be included in the PATH variable for experiments"]
+
+* Loading other experiments or projects can be specifying similar to python imports
```
Comment (cormensratio) [inline docs/source/vara-ts/benchbuild.rst]:
```suggestion
* Loading other experiments or projects can be specified similar to python imports
```

File: docs/source/vara-ts/benchbuild.rst [hunk 6]
```diff
@@ -22,3 +22,57 @@ Second, we change into the benchbuild folder and run an experiment that generate
...
+
+.. code-block:: yaml
+
+  versions:
+    full:
+      default: false
+      desc: Ignore default sampling and provide full version exploration.
+      value: true
+
+* Adapt the number of threads that should be for running project compilation.
```
Comment (cormensratio) [inline docs/source/vara-ts/benchbuild.rst]:
```suggestion
* Adapt the number of threads that should be running for project compilation.
```

File: docs/source/research_tool_docs/vara/debugging.rst [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+.. code-block:: llvm
+
+  ; RUN: %clang ... %test_repos/BasicTestRepos/ExampleRepo/main.c ...
+  ; RUN: %opt ... %test_repos/BasicTestRepos/ExampleRepo/ ...
+
+
+Debugging options
+=================
+VaRA can produce specific debug output to ease debugging, to enable this output use LLVM debugging options.
```
Comment (cormensratio) [inline docs/source/research_tool_docs/vara/debugging.rst]:
```suggestion
VaRA can produce specific debug output to ease debugging. To enable this output use LLVM debugging options.
```

**Review by cormensratio — COMMENTED** [review]
lgtm

**Comment by cormensratio** [comment]
In the Tool-suite overview section, the LLVM Logo and VaRA Logo is slightly overlapping.

File: docs/source/index.rst (hunk 1 above)
Comment (vulder) [inline docs/source/index.rst]:
```suggestion
Tool-Suite Overview
```

File: docs/source/index.rst [hunk 8]
```diff
@@ -8,20 +8,24 @@ Welcome to VaRA's documentation!
...
 
 .. toctree::
    :maxdepth: 1
    :caption: List of provided research tools:
 
-   vara-ts-api/tools/research_tools/vara
+   research_tool_docs/vara/vara
+
+
+Tool-Suite overview
```
Comment (vulder) [inline docs/source/index.rst]:
```suggestion
Tool-Suite Overview
```

**Comment by vulder** [comment]
> In the Tool-suite overview section, the LLVM Logo and VaRA Logo is slightly overlapping.

This is intended. The idea is to show that VaRA is integrated into llvm.

**Comment by boehmseb** [comment]
What I find a bit annoying is that the VaRA logo behind the LLVM logo has a different aspect ratio, i.e. it is stretched.

**Comment by boehmseb** [comment]
Also, the contrast between the logos and the background in this box is quite bad and creates noise around the edges of the LLVM logo.

File: docs/source/research_tool_docs/vara/debugging.rst [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+To run all llvm related regression tests in ``test/vara-llvm/``:
+
+.. code-block:: bash
+
+  make check-vara-llvm
+
+Testing commit and git analyses
+-------------------------------
+
+For testing of commit and git related analyses, VaRA offers additional testing repositories, which are automatically provided to your regression or unittests.
```
Comment (boehmseb) [inline docs/source/research_tool_docs/vara/debugging.rst]:
```suggestion
For testing of commit and git related analyses, VaRA offers additional testing repositories which are automatically provided to your regression or unittests.
```

File: docs/source/research_tool_docs/vara/debugging.rst [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+For unittests, just include the ``UnittestHelper.h`` and load a repository by specifying it's subpath, relative to the ``TestRepositories`` folder.
```
Comment (boehmseb) [inline docs/source/research_tool_docs/vara/debugging.rst]:
```suggestion
For unittests, just include the ``UnittestHelper.h`` and load a repository by specifying its subpath, relative to the ``TestRepositories`` folder.
```

File: docs/source/research_tool_docs/vara/debugging.rst [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+.. code-block:: llvm
+
+  ; RUN: %clang ... %test_repos/BasicTestRepos/ExampleRepo/main.c ...
+  ; RUN: %opt ... %test_repos/BasicTestRepos/ExampleRepo/ ...
+
+
+Debugging options
+=================
+VaRA can produce specific debug output to ease debugging. To enable this output use LLVM debugging options.
+Add the corresponding debug flag to your run line `-debug-only=$DB_FLAG`
```
Comment (boehmseb) [inline docs/source/research_tool_docs/vara/debugging.rst]:
```suggestion
Add the corresponding debug flag to your run line ``-debug-only=$DB_FLAG``
```

File: docs/source/research_tool_docs/vara/setup.rst [hunk 12]
```diff
@@ -0,0 +1,62 @@
+How to setup VaRA
+=================
+This is a quick guide on how to setup VaRA either with our automatic setup or manually.
+
+Manual Installation
+-------------------
+VaRA can be automatically set up with :ref:`vara-buildsetup` from the tool suite, for instructions see setup.
```
Comment (boehmseb) [inline docs/source/research_tool_docs/vara/setup.rst]: Missing link `setup`

File: docs/source/research_tool_docs/vara/setup.rst [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+If you want to manually install llvm and VaRA see the following instructions.
+
+Linux
+.....
+To create your own setup or integrate VaRA into LLVM follow these instructions.
```
Comment (boehmseb) [inline docs/source/research_tool_docs/vara/setup.rst]:
```suggestion
To create your own setup or integrate VaRA into LLVM, follow these instructions.
```

File: docs/source/vara-ts/benchbuild.rst [hunk 14]
```diff
@@ -22,3 +22,57 @@ Second, we change into the benchbuild folder and run an experiment that generate
   benchbuild -vv run -E GenerateBlameReport gzip
 
 The generated result files are place in the ``vara/results/$PROJECT_NAME`` folder and can be further visualized with VaRA-TS graph generators.
+
+How-to configure BenchBuild yourself
+------------------------------------
+BenchBuild's configuration file ``.benchbuild.yml`` normally is placed inside the `benchbuild` folder, which is located in the vara root folder.
```
Comment (boehmseb) [inline docs/source/vara-ts/benchbuild.rst]:
```suggestion
BenchBuild's configuration file ``.benchbuild.yml`` normally is placed inside the ``benchbuild`` folder, which is located in the vara root folder.
```

File: docs/source/vara-ts/benchbuild.rst [hunk 15]
```diff
@@ -22,3 +22,57 @@ Second, we change into the benchbuild folder and run an experiment that generate
...
+A default version of this file can be automatically generated with our tool :ref:`vara-gen-bbconfig`.
+To adapt and tune BenchBuild further a lot of different configuration flags can be adapted. The following list shows the most important ones:
```
Comment (boehmseb) [inline docs/source/vara-ts/benchbuild.rst]:
```suggestion
To adapt and tune BenchBuild further, you can moify the different configuration flags in this config file. The following list shows the most important ones:
```

File: docs/source/vara-ts/benchbuild.rst [hunk 16]
```diff
@@ -22,3 +22,57 @@ Second, we change into the benchbuild folder and run an experiment that generate
...
+
+* Adding extra paths to the environment
+
+.. code-block:: yaml
+
+  env:
+    path:
+      value: ["paths from your system that should be included in the PATH variable for experiments"]
+
+* Loading other experiments or projects can be specified similar to python imports
```
Comment (boehmseb) [inline docs/source/vara-ts/benchbuild.rst]:
```suggestion
* Other experiments or projects can be loaded similar to python imports
```

File: docs/source/vara-ts/benchbuild.rst [hunk 17]
```diff
@@ -22,3 +22,57 @@ Second, we change into the benchbuild folder and run an experiment that generate
...
+
+.. code-block:: yaml
+
+  versions:
+    full:
+      default: false
+      desc: Ignore default sampling and provide full version exploration.
+      value: true
+
+* Adapt the number of threads that should be running for project compilation.
```
Comment (boehmseb) [inline docs/source/vara-ts/benchbuild.rst]:
```suggestion
* Adapt the number of threads that should be used for project compilation.
```

**Review by boehmseb — CHANGES_REQUESTED** [review]

**Review by boehmseb — APPROVED** [review]

**Pull request merged**